src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.ts

```diff
@@ -0,0 +1,144 @@
+import { createRequestSender } from '@bigcommerce/request-sender';
+import { round } from 'lodash';
+
+import { Checkout } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { ContentType } from '../../../common/http-request';
+import { PaymentMethodFailedError } from '../../errors';
+import PaymentMethod from '../../payment-method';
+import { CheckoutcomGooglePayToken, CheckoutcomToken } from '../checkoutcom';
+
+import { BillingAddressFormat, GooglePaymentData, GooglePayInitializer, GooglePayPaymentDataRequestV2, TokenizePayload } from './googlepay';
+
+export default class GooglePayCheckoutcomInitializer implements GooglePayInitializer {
+
+    private publishableKey: string = '';
+    private testMode: boolean = true;
+
+    async initialize(
+        checkout: Checkout,
+        paymentMethod: PaymentMethod,
+        hasShippingAddress: boolean
+    ): Promise<GooglePayPaymentDataRequestV2> {
+        this.publishableKey = paymentMethod.initializationData.checkoutcomkey;
+        this.testMode = !!paymentMethod.config.testMode;
+
+        return this._mapGooglePayCheckoutcomDataRequestToGooglePayDataRequestV2(
+            checkout,
+            paymentMethod.initializationData,
+            hasShippingAddress
+        );
+    }
+
+    teardown(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    async parseResponse(paymentData: GooglePaymentData): Promise<TokenizePayload> {
+        let token;
+        try {
+            token = JSON.parse(paymentData.paymentMethodData.tokenizationData.token);
+        } catch (err) {
+            throw new InvalidArgumentError('Unable to parse response from Google Pay.');
+        }
+
+        if (!token.signature || !token.protocolVersion || !token.signedMessage) {
+            throw new PaymentMethodFailedError('Unable to parse response from Google Pay.');
+        }
+        const finalToken = await this._convertToken(this.testMode, this.publishableKey, token);
+
+        return Promise.resolve({
+            nonce: finalToken,
+            type: 'CreditCard',
+            description: paymentData.paymentMethodData.description,
+            details: {
+                cardType: paymentData.paymentMethodData.info.cardNetwork,
+                lastFour: paymentData.paymentMethodData.info.cardDetails,
+            },
+        });
+    }
+
+    private _convertToken(testMode: boolean, checkoutcomkey: string, token: CheckoutcomGooglePayToken): Promise<string> {
+        if (!token || !token.protocolVersion) {
+            throw new PaymentMethodFailedError('Unable to parse response from GooglePay.');
+        }
+
+        return this._requestCheckoutcomTokenize(testMode, checkoutcomkey, {
```

Comment:
Please use `async/await` in this file. We don't want a new file to have a mixture of `async/await` vs `then`.

Reply:
Changed to async/await